consensus/reactor.go

```diff
@@ -1375,6 +1383,23 @@ type NewRoundStepMessage struct {
 	LastCommitRound       int
 }
 
+// ValidateBasic performs basic validation.
+func (m *NewRoundStepMessage) ValidateBasic() error {
+	if m.Height < 0 {
+		return errors.New("Negative Height")
+	}
+	if m.Round < 0 {
+		return errors.New("Negative Round")
+	}
+	if m.SecondsSinceStartTime < 0 {
+		return errors.New("Negative SecondsSinceStartTime")
+	}
+	if m.LastCommitRound < -1 {
```

Comment:
```
2018-10-31T10:59:12+00:00 172.17.0.1 local_testnet_2[1256]: E[31106-10-31|10:59:12.176] Peer sent us invalid msg                     module=consensus peer="Peer{MConn{172.5
7.0.101:38716} 55854e179f195424fbf969016e1a2ab47e1a0aea in}" msg="[NewRoundStep H:22 R:0 S:RoundStepPrevote LCR:-1]" err="Invalid LastCommitRound (for 1st block: -1, for ot
hers: >= 0)"
2018-10-31T10:59:12+00:00 172.17.0.1 local_testnet_2[1256]: E[31106-10-31|10:59:12.176] Stopping peer for error                      module=p2p peer="Peer{MConn{172.57.0.10
1:38716} 55854e179f195424fbf969016e1a2ab47e1a0aea in}" err="Invalid LastCommitRound (for 1st block: -1, for others: >= 0)"
```

I am observing a different picture. p2p tests are failing because of this